### DISCUSSION TITLE
Strong consistency like Amazon S3

### DISCUSSION THREAD

**Discussion by immanuelfodor** [minio/minio#11389] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

Amazon has just announced strong consistency for S3. Will Minio support this in the future? If yes, what is the priority on this feature? 

https://aws.amazon.com/s3/consistency/

**Describe the solution you'd like**

> After a successful write of a new object or an overwrite of an existing object, any subsequent read request immediately receives the latest version of the object. S3 also provides strong consistency for list operations, so after a write, you can immediately perform a listing of the objects in a bucket with any changes reflected.

**Describe alternatives you've considered**

> Without strong consistency, you would insert custom code into these applications, or provision databases to keep objects consistent with any changes in Amazon S3 across millions or billions of objects.

**Additional context**

HackerNews thread: https://news.ycombinator.com/item?id=25271791

**Comment by immanuelfodor** [comment]
I might have just found the answer, I just needed to search for strict instead of strong: https://github.com/minio/minio/blob/master/docs/distributed/README.md#consistency-guarantees

> MinIO follows strict read-after-write and list-after-write consistency model for all i/o operations both in distributed and standalone modes.

This seems to be the same for me to what Amazon offers now. Is it?

**Comment by harshavardhana** [comment, marked as answer]
> This seems to be the same for me to what Amazon offers now. Is it?

MinIO has always been strictly consistent.

**Comment by immanuelfodor** [comment]
Awesome, thank you!
